decidim-core/app/helpers/decidim/amendments_helper.rb

```diff
@@ -56,24 +56,59 @@ def emendation_announcement_for(emendation)
       cell "decidim/amendable/announcement", emendation if emendation.emendation?
     end
 
-    # Renders the buttons to accept/reject an emendation (for amendable authors)
+    # Returns Html action button cards: accept/reject or promote
     #
     # emendation - The resource that is an emendation.
-    #
-    # Returns Html action button card
-    def emendation_actions_for(emendation_form)
-      cell("decidim/amendable/emendation_actions", emendation_form).to_s if emendation_form.emendation.emendation? && allowed_to_react_to_emendation?(emendation_form)
+    def emendation_actions_for(emendation)
+      return unless allowed_to_react_to_emendation?(emendation)
+      return accept_and_reject_buttons_for(emendation) if allowed_to_accept_and_reject?(emendation)
+      return promote_button_for(emendation) if allowed_to_promote?(emendation)
+    end
+
+    # Renders the buttons to accept/reject an emendation
+    def accept_and_reject_buttons_for(emendation)
+      cell("decidim/amendable/emendation_actions", emendation)
+    end
+
+    # Renders the button to promote an emendation
+    def promote_button_for(emendation)
+      cell("decidim/amendable/promote_button_card", emendation)
     end
 
     # Checks if current_user can react to the emendation
     #
     # Returns true or false
-    def allowed_to_react_to_emendation?(emendation_form)
-      if current_user && current_user.active_role.present?
-        return true if current_user.active_role.include?("admin")
-      end
+    def allowed_to_react_to_emendation?(emendation)
+      return unless emendation.emendation?
+      return unless current_user
+      true
+    end
 
-      emendation_form.amendable.created_by?(current_user)
+    # Checks if current_user can accept and reject the emendation
+    #
+    # Returns true or false
+    def allowed_to_accept_and_reject?(emendation)
+      return unless emendation.amendment.evaluating?
+      emendation.amendable.created_by?(current_user) || current_user.admin
+    end
+
+    # Checks if current_user can promote the emendation
+    #
+    # Returns true or false
+    def allowed_to_promote?(emendation)
+      return unless emendation.state == "rejected"
+      return unless emendation.created_by?(current_user)
+      not_yet_promoted(emendation)
+    end
+
+    # Checks whether the ActionLog created in the promote command exists.
+    #
+    # Returns true or false
+    def not_yet_promoted(emendation)
+      logs = Decidim::ActionLog.where(decidim_component_id: emendation.component)
+                               .where(decidim_user_id: emendation.creator_author)
```

Comment:
I might have asked this already but: what can the emendation creator author not be a user? If so, we should check something here.

Reply:
The log we are looking for is the one created in the promote command.
https://github.com/decidim/decidim/blob/72fdb7f19845ee4cb16cba9af33222e49e688a0c/decidim-core/app/commands/decidim/amendable/promote.rb#L36
The user that performs the action of the log can only be one: the current user at the moment of promoting the emendation, which is the emendation creator author.

So, we are searching all the logs created in the current component by the current user with action: "promote", which can be more than one if the user has promoted many rejected emendations. Once we have this array, we select the one that matches the extra piece of information introduced in the log: "promoted_from" => id.

Reply:
I mean that since in other entities authors can be other things than users (organizations, meetings, etc) maybe this also applies here.

Reply:
As of now, amendments can only be created from the public view, which means the amender will always be a user (admin or not). When the amendment is rejected and the opportunity to promote arises, the same user that created the amendment will be the one that promotes it, which means that the traceability action will be performed by this user.

All in all, we are looking for an ActionLog to check if the emendation has already been promoted and entering the user that created the log narrows it down, nothing more.